From e514f8e783f46ee56d96e444fab651d60ce526d2 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Mon, 3 Aug 2020 21:00:53 -0700
Subject: [PATCH 729/830] ANDROID: GKI: enable RCU configs
X-NVConfidentiality: public

Enable configs needed for RCU perf/power optimizations

CONFIG_RCU_EXPERT=y
CONFIG_RCU_FAST_NO_HZ=y
CONFIG_RCU_NOCB_CPU=y

Bug: 162753392
Signed-off-by: Todd Kjos <tkjos@google.com>
Change-Id: Ie261e9dbd679c2f2481ebc1187c0ac4a01fd0767
---
 arch/arm64/configs/gki_defconfig | 3 +++
 arch/x86/configs/gki_defconfig   | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index fc5c3c2709d3..6a354201791e 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -8,6 +8,9 @@ CONFIG_TASKSTATS=y
 CONFIG_TASK_XACCT=y
 CONFIG_TASK_IO_ACCOUNTING=y
 CONFIG_PSI=y
+CONFIG_RCU_EXPERT=y
+CONFIG_RCU_FAST_NO_HZ=y
+CONFIG_RCU_NOCB_CPU=y
 CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
 CONFIG_IKHEADERS=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index bc5cac9f674d..ac097afb5695 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -10,6 +10,9 @@ CONFIG_TASKSTATS=y
 CONFIG_TASK_XACCT=y
 CONFIG_TASK_IO_ACCOUNTING=y
 CONFIG_PSI=y
+CONFIG_RCU_EXPERT=y
+CONFIG_RCU_FAST_NO_HZ=y
+CONFIG_RCU_NOCB_CPU=y
 CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
 CONFIG_IKHEADERS=y
-- 
2.17.1

